src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx

```diff
@@ -0,0 +1,125 @@
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+const FOUR_PIECE_BONUS = 0.4;
+const FOUR_PIECE_SPELLS = [SPELLS.RENEWING_MIST_HEAL, SPELLS.VIVIFY];
+
+class T30TierSet extends Analyzer {
+  has2Piece: boolean = true;
+  has4Piece: boolean = true;
+  vivHealing: number = 0;
+  renewingMistHealing: number = 0;
+  manaGain: number = 0;
+  wastedManaGain: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.has2Piece = this.selectedCombatant.has2PieceByTier(TIERS.T29);
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T29) && this.has2Piece;
+    this.active = this.has2Piece;
+    if (!this.active) {
+      return;
+    }
+    // 2pc do energized for mana gain value and show uptime
+    this.addEventListener(
+      Events.resourcechange.by(SELECTED_PLAYER).spell(SPELLS.SOULFANG_INFUSION),
+      this.handle2pcMana,
+    );
+    //4pc do uptime and effective healing increase
+    if (this.has4Piece) {
+      this.addEventListener(
+        Events.heal.by(SELECTED_PLAYER).spell(FOUR_PIECE_SPELLS),
+        this.handle4PcHeal,
+      );
+    }
+  }
+
+  get total4PieceHealing() {
+    return this.renewingMistHealing + this.vivHealing;
+  }
+  get twoSetUptime() {
+    return (
+      this.selectedCombatant.getBuffUptime(SPELLS.SOULFANG_INFUSION.id) / this.owner.fightDuration
+    );
+  }
+
+  get fourSetUptime() {
+    return (
+      this.selectedCombatant.getBuffUptime(SPELLS.SOULFANG_VITALITY.id) / this.owner.fightDuration
+    );
+  }
+
+  handle2pcMana(event: ResourceChangeEvent) {
+    if (event.resourceChangeType !== RESOURCE_TYPES.MANA.id) {
+      return;
+    }
+    this.manaGain += event.resourceChange - event.waste;
+    this.wastedManaGain += event.waste;
+  }
+
+  handle4PcHeal(event: HealEvent) {
+    const spellId = event.ability.guid;
+    if (this.selectedCombatant.hasBuff(SPELLS.SOULFANG_VITALITY.id)) {
+      if (spellId === SPELLS.RENEWING_MIST_HEAL.id) {
+        this.renewingMistHealing += calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
+      } else if (spellId === SPELLS.VIVIFY.id) {
+        this.vivHealing += calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
+      }
+    }
+  }
+
+  statistic() {
+    return (
+      <Statistic
+        size="flexible"
+        position={STATISTIC_ORDER.OPTIONAL(0)}
+        category={STATISTIC_CATEGORY.ITEMS}
+        tooltip={
+          <ul>
+            <li>
+              <strong>{formatNumber(this.renewingMistHealing)}</strong> extra{' '}
+              <SpellLink id={SPELLS.RENEWING_MIST_HEAL.id} /> healing
+            </li>
+            <li>
+              <strong>{formatNumber(this.vivHealing)}</strong> extra{' '}
+              <SpellLink id={SPELLS.VIVIFY.id} /> healing
+            </li>
+            <li>
+              <strong>{formatNumber(this.wastedManaGain)}</strong> mana wasted from overcapping
+            </li>
+          </ul>
+        }
+      >
+        <BoringValueText label="T29 Tier Set">
```

Comment:
T30? Alternatively you could name it after https://www.wowhead.com/ptr/item-set=1546/fangs-of-forged-vermillion

Reply:
🤦‍♂️